Fix Thunderbird Autoconfiguration redirects

mss opened the pull request:
<!-- 🙌 Thanks for contributing to MDN Web Docs. Adding details below will help us to merge your PR faster. -->

### Description

<!-- ✍️ Summarize your changes in one or two sentences -->

Ad some point (probably between #1853 and #4680) the canonical URL for the Thunderbird Autoconfiguration spec was changed from `/en-US/docs/Thunderbird/Autoconfiguration` to `/en-US/docs/Mozilla/Thunderbird/Autoconfiguration`. Due to that both pages are linked nowadays but the existing redirect takes care of the first only.

### Motivation

<!-- ❓ Why are you making these changes and how do they help readers? -->

Links to the specification are broken all over the internet.

### Additional details

<!-- 🔗 Link to release notes, vendor docs, bug trackers, source control, or other places providing more context -->

Since this seems to be the canonical source for the specification which is implemented by Thunderbird and some other applications and I think the content should actually be unarchived. But this change improves the current situation. 

### Related issues and pull requests

<!-- 🔨 If this fully resolves a GitHub issue, use "Fixes #123" -->
<!-- 👉 Highlight related pull requests using "Relates to #123" -->
<!-- ❗ If another pull request should be merged first, use "**Depends on:** #123" -->


<!-- 👷‍♀️ After submitting, go to the "Checks" tab of your PR for the build status -->

mss committed:
Fix Thunderbird Autoconfiguration redirects (#28783)

mss commented:
Actually, it seems like the canonical spec is still at https://wiki.mozilla.org/Thunderbird:Autoconfiguration:ConfigFileFormat and was just linked from here. But the spec page itself below `Description` links to two different formats of these URLs (the first one is still broken after this change).

mss commented:
@benbucksch maybe these pages should be moved back to the Mozilla Wiki and these redirect point there instead?

mss commented:
I fixed the main reason for the `check-redirects` workflow (I had to manually edit the file die to #28782 and didn't sort it properly) but the check still triggers because I introduced some redirect chain by restoring the old canonicalization of the legacy URL. IMO this is a false positive but it could be avoided by just not doing that if wanted.

benbucksch commented:
@mss Thanks for caring about these docs.

The current version of the documents are at https://www.bucksch.org/1/projects/thunderbird/autoconfiguration/
You may redirect there.

I plan to create a github repo and maybe move the documents there into that Wiki. I could then adjust the links on my server to that new place.

benbucksch committed:
Update Thunderbird Autoconfiguration redirects (#28783)

mss commented:
@benbucksch I added a commit to update the redirects accordingly. I think the MDN redirects should be updated again once you moved the content but for now this should ensure that legacy links point to the latest spec.

benbucksch commented:
Thanks for the redirect. That's great!

I've checked the links in the patch, and they are correct. r+ from my side.

chrisdavidmills commented:
Thanks for the contribution, @mss. I've just run the lining tests, and unfortunately, the check redirects tests are failing with a not-very-helpful error message. From my experience, the MDN redirects system is brittle and fiddly.

Some thoughts:

- Did you change the file manually? I was always told to never change the file manually, and use the `add-redirect` tool instead: https://github.com/mdn/content/blob/main/CONTRIBUTING.md#redirecting-a-document
- I'm assuming you made sure that the syntax pattern matches exactly the other lines in the file. This includes making sure the separator character between the from and to URLs is the same.

If my limited wisdom does not help, we might have to call in one of the MDN devs to offer advice.

mss commented:
@chrisdavidmills thanks for checking. I originally had bug due to manual editing (cf. my comment from 2 days ago) but I fixed that.

Now I think my change is ok but I hit an issue in the code behing `yarn content validate-redirects en-us --strict` which doesn't support redirect chains. After my change we have the chain of `/en-US/docs/Thunderbird/Autoconfiguration` `->` `/en-US/docs/Mozilla/Thunderbird/Autoconfiguration` `->` `https://www.bucksch.org/1/projects/thunderbird/autoconfiguration/`. And that's exactly what makes the `--strict` mode of the test  (which recursively resolves the redirect chain and bails out if the target of that resolution is different to the value in the file) to fail.

I actually don't understand why that strict mode is called in the first place (comments would be nice), my guess is that it is supposed to detect redirect loops. If that's the case it should just stop at external redirects. I can add that but have no idea if that's the intention. So some dev who knows that yari thingy has to have a look indeed.

After quite some yak shaving I think there are two possible solutions:
* Duplicate the redirect targets in the file for both source URLs. I personally hate that but then no redirect chain would exist.
* Fix that yari tool to handle this case.

mss committed:
Duplicate target URLs to avoid redirect chains (#28783)

mss commented:
I rebased on `main` and added an additional comment which avoids the redirect chains. Now the `--strict` mode doesn't complain on my machine. So this should be mergeable but if a better solution is found the last commit can just be dropped.

chrisdavidmills commented:
I'll get this merged so that the value is seen immediately, but I'll also bring this to the attention of our devs. thanks @mss .

@fiji-flo @LeoMcA see https://github.com/mdn/content/pull/28783#issuecomment-1702501713 for an interesting comment on the redirect linting. Might be worth giving some consideration to.

chrisdavidmills reviewed (approved):
LGTM
